In [1]:
# hou_prd_details.csv
# housewarming2.csv

import pandas as pd

prd = pd.read_csv("hou_prd_details.csv")
house = pd.read_csv("housewarming2.csv")

prd = prd.drop('Unnamed: 0', axis = 1)
# house = house.drop('Unnamed: 0', axis = 1)

prd.head(10)

id                                        상품url    브랜드명  \
0   0   https://ohou.se/productions/538782/selling     에어론   
1   0   https://ohou.se/productions/803775/selling    베네베네   
2   0  https://ohou.se/productions/1068049/selling   플렌스테드   
3   0   https://ohou.se/productions/580038/selling     잠솔솔   
4   0   https://ohou.se/productions/961600/selling      리스   
5   0   https://ohou.se/productions/357692/selling     마켓비   
6   0  https://ohou.se/productions/1137934/selling    LG전자   
7   0  https://ohou.se/productions/1006108/selling    스펙트라   
8   0  https://ohou.se/productions/1438532/selling  touffu   
9   0  https://ohou.se/productions/1539603/selling   에피큐리언   

                                       상품명  \
0  리버서블 오피스 데스크 책상/테이블 600~2400mm(2colors)   
1                 벤트라이크 멀티플 접이식 세발자전거+안전가드   
2    FLENSTED MOBILES 모빌 - 천사 Angel mobile   
3                경추목 메모리폼 숙면 꿀잠 수면 베개_항균탠샐   
4                   리스 클래식 법랑 양수냄비 22cm(H)   
5                   PRISA 펜던트등 E26 KS1733D   
6         LG 디오스 식기세척기 오브제컬렉션 DUBJ2CA 12인용   
7              [키즈페어] UV-LED 젖병소독기 2colors   
8                           cloud mirror 2   
9    [리퍼] A마존 1위도마! 미국제조 건강한도마 대+핸디size 세트   

                                   카테고리       상품가격     리뷰수     스크랩수  스타일링샷수  
0     ['가구', '테이블·식탁·책상', '책상', '일반책상']    39800.0  2891.0  57140.0  2113.0  
1         ['유아·아동', '완구·교구', '공간·대형완구']    94800.0     1.0     60.0     1.0  
2         ['데코·식물', '인테리어소품', '모빌·가랜드']    37400.0     NaN     81.0     0.0  
3             ['패브릭', '베개·베개커버', '폼베개']    38900.0    32.0     78.0    13.0  
4       ['주방용품', '냄비·프라이팬·솥', '냄비·뚝배기']   135000.0     1.0     89.0     1.0  
5                ['조명', '천장등', '펜던트조명']    36900.0     3.0    100.0     5.0  
6  ['가전', '주방가전', '식기세척기·건조기', '식기세척기']  1699000.0     8.0    116.0    62.0  
7              ['가전', '주방가전', '기타주방가전']   268900.0    11.0    127.0    26.0  
8                  ['가구', '거울', '탁상거울']    50000.0     NaN    153.0     0.0  
9     ['주방용품', '조리도구·도마', '도마', '나무도마']    43800.0    17.0    163.0    12.0

In [2]:
house.head()

id         작성일                              제목          가족형태   공간  평수구간  \
0   0  2022-10-07  서재형 거실부터 베란다 아뜰리에까지! 센스만점 오베네홈      아기가 있는 집  아파트    35   
1   1  2022-10-07  구축 2Bay 리모델링의 바이블, 모던한 32평 신혼집          신혼부부  아파트    35   
2   2  2022-10-07    취향을 읽고 쓰는 비밀 서재처럼- 푸릇푸릇 3평 방  부모님과 함께 사는 집   기타     5   
3   3  2022-10-07   손 뻗으면 닿을 듯한 녹음🌳 싱그러운 저층집 리모델링          신혼부부  아파트    35   
4   4  2022-10-07  아늑하고 편안한 무드로- 빈티지 한 스푼 더한 내추럴홈          신혼부부  아파트    55   

   스타일                              세부공사     분야      작업    예산     조회  스크랩  \
0  내추럴                                 0   리모델링     전문가     0  10513  136   
1   모던  주방리모델링, 조명시공, 중문, 가벽&파티션, 슬라이딩도어   리모델링     전문가  4200   5610   75   
2  내추럴                                 0  홈스타일링  셀프•DIY     0   4214  201   
3   모던                  주방리모델링, 조명시공, 중문   리모델링     전문가  5500  37375  643   
4  내추럴                        주방리모델링, 중문   리모델링     전문가  9500   3760   73   

   좋아요   댓글                                                url  
0   57   15  https://ohou.se/projects/126917/detail?affect_...  
1   26   12  https://ohou.se/projects/126600/detail?affect_...  
2   60   15  https://ohou.se/projects/123056/detail?affect_...  
3  169  122  https://ohou.se/projects/116945/detail?affect_...  
4   13    9  https://ohou.se/projects/127436/detail?affect_...

In [54]:
prd.isnull().sum()

id            0
상품url         0
브랜드명          0
상품명           0
카테고리          0
상품가격      78943
리뷰수       88274
스크랩수        323
스타일링샷수       31
dtype: int64

In [17]:
prd['카테고리'].apply(lambda x: x.split(','))

0         [['가구',  '테이블·식탁·책상',  '책상',  '일반책상']]
1              [['유아·아동',  '완구·교구',  '공간·대형완구']]
2              [['데코·식물',  '인테리어소품',  '모빌·가랜드']]
3                  [['패브릭',  '베개·베개커버',  '폼베개']]
4            [['주방용품',  '냄비·프라이팬·솥',  '냄비·뚝배기']]
                           ...                  
243014               [['데코·식물',  '홈갤러리',  '액자']]
243015          [['패브릭',  '러그·카페트',  '라탄·수초러그']]
243016           [['데코·식물',  '인테리어소품',  '장식소품']]
243017                 [['가구',  '의자',  '스툴·벤치']]
243018       [['가구',  '행거·옷장',  '행거',  '스탠드행거']]
Name: 카테고리, Length: 243019, dtype: object

In [46]:
pd.set_option('display.max_rows',None)
category_gp = pd.DataFrame(prd['카테고리'].value_counts())
category_gp = category_gp.reset_index()
category_gp = category_gp.rename(columns = {'index':'category','카테고리':'cnt'})
category_gp.sort_values('category')

category   cnt
718                       ['가구', '가벽·파티션', '사무용파티션']    10
226                      ['가구', '가벽·파티션', '인테리어파티션']   255
251                       ['가구', '거실장·TV장', 'TV스탠드']   215
200                 ['가구', '거실장·TV장', '높은거실장·사이드보드']   305
121                       ['가구', '거실장·TV장', '일반거실장']   551
22                               ['가구', '거울', '벽거울']  1802
51                              ['가구', '거울', '전신거울']  1230
56                              ['가구', '거울', '탁상거울']  1131
625             ['가구', '매트리스·토퍼', '매트리스', '라텍스매트리스']    20
141             ['가구', '매트리스·토퍼', '매트리스', '스프링매트리스']   459
254               ['가구', '매트리스·토퍼', '매트리스', '폼매트리스']   213
416                          ['가구', '매트리스·토퍼', '토퍼']    85
23                           ['가구', '서랍·수납장', '서랍장']  1718
10                           ['가구', '서랍·수납장', '수납장']  2554
578                 ['가구', '서랍·수납장', '주방수납장', '그릇장']    29
321                 ['가구', '서랍·수납장', '주방수납장', '렌지대']   148
455                 ['가구', '서랍·수납장', '주방수납장', '상부장']    65
362             ['가구', '서랍·수납장', '주방수납장', '상부장+하부장']   123
425                 ['가구', '서랍·수납장', '주방수납장', '키큰장']    81
292                 ['가구', '서랍·수납장', '주방수납장', '하부장']   176
296                          ['가구', '서랍·수납장', '캐비닛']   172
59                            ['가구', '서랍·수납장', '협탁']  1115
132                     ['가구', '선반', '벽선반', '무지주선반']   496
74                       ['가구', '선반', '벽선반', '지주선반']   825
184                      ['가구', '선반', '벽선반', '찬넬선반']   333
28                             ['가구', '선반', '스탠드선반']  1659
224                         ['가구', '선반', '앵글·조립식선반']   258
273                            ['가구', '소파', '리클라이너']   193
342                             ['가구', '소파', '소파베드']   134
481                             ['가구', '소파', '소파스툴']    55
7                               ['가구', '소파', '일반소파']  3113
508                             ['가구', '소파', '좌식소파']    45
495                            ['가구', '의자', '게이밍의자']    50
160                              ['가구', '의자', '바체어']   378
542                              ['가구', '의자', '발받침']    37
329                               ['가구', '의자', '빈백']   142
8                              ['가구', '의자', '스툴·벤치']  2739
149                             ['가구', '의자', '안락의자']   424
0                             ['가구', '의자', '인테리어의자']  6643
540                      ['가구', '의자', '좌식의자·자세보정의자']    38
81                          ['가구', '의자', '학생·사무용의자']   774
493                             ['가구', '의자', '흔들의자']    51
257                         ['가구', '진열장·책장', '매거진랙']   211
75                       ['가구', '진열장·책장', '진열장·장식장']   818
24                            ['가구', '진열장·책장', '책장']  1710
630                  ['가구', '침대', '침대+매트리스', '모션베드']    19
344                  ['가구', '침대', '침대+매트리스', '수납침대']   132
581               ['가구', '침대', '침대+매트리스', '이층·벙커침대']    29
93                   ['가구', '침대', '침대+매트리스', '일반침대']   692
565                 ['가구', '침대', '침대+매트리스', '저상형침대']    31
608                 ['가구', '침대', '침대+매트리스', '투매트리스']    24
535                 ['가구', '침대', '침대+매트리스', '패밀리침대']    38
319                           ['가구', '침대', '침대부속가구']   152
814                 ['가구', '침대', '침대프레임', '돌침대·흙침대']     5
622                    ['가구', '침대', '침대프레임', '모션베드']    20
241                    ['가구', '침대', '침대프레임', '수납침대']   228
689                 ['가구', '침대', '침대프레임', '이층·벙커침대']    13
72                     ['가구', '침대', '침대프레임', '일반침대']   868
247                   ['가구', '침대', '침대프레임', '저상형침대']   220
891                   ['가구', '침대', '침대프레임', '접이식침대']     1
724                   ['가구', '침대', '침대프레임', '패밀리침대']    10
490            ['가구', '침대', '침대프레임', '하단매트리스·파운데이션']    52
34                   ['가구', '테이블·식탁·책상', '거실·소파테이블']  1469
13                     ['가구', '테이블·식탁·책상', '사이드테이블']  2284
167               ['가구', '테이블·식탁·책상', '식탁', '식탁+의자']   364
3              ['가구', '테이블·식탁·책상', '식탁', '식탁·입식테이블']  3606
368           ['가구', '테이블·식탁·책상', '식탁', '홈바·아일랜드식탁']   120
576              ['가구', '테이블·식탁·

In [50]:
len(prd['카테고리'].apply(lambda x : x.split(',')[0]).unique())

19

In [2]:
house.head(10)

id         작성일                              제목          가족형태       공간  \
0   0  2022-10-07  서재형 거실부터 베란다 아뜰리에까지! 센스만점 오베네홈      아기가 있는 집      아파트   
1   1  2022-10-07  구축 2Bay 리모델링의 바이블, 모던한 32평 신혼집          신혼부부      아파트   
2   2  2022-10-07    취향을 읽고 쓰는 비밀 서재처럼- 푸릇푸릇 3평 방  부모님과 함께 사는 집       기타   
3   3  2022-10-07   손 뻗으면 닿을 듯한 녹음🌳 싱그러운 저층집 리모델링          신혼부부      아파트   
4   4  2022-10-07  아늑하고 편안한 무드로- 빈티지 한 스푼 더한 내추럴홈          신혼부부      아파트   
5   5  2022-10-07    게으른 집순이가 폭신한 하루를 보내는 복층 오피스텔         싱글라이프  원룸&오피스텔   
6   6  2022-10-07  사계절을 품은 액자창이 포인트! 따뜻한 무드의 신혼집           신혼부부      아파트   
7   7  2022-10-07  알록달록은 꼭꼭 숨겨라! 아기와 함께 하는 베이지하우스      아기가 있는 집      아파트   
8   8  2022-10-07   헌집줄게 새집다오🏘 살던 집을 리모델링 해 새집처럼!      아기가 있는 집      아파트   
9   9  2022-10-07      자취를 하고 나서 퇴근 후의 삶이 행복해졌어요💗         싱글라이프  원룸&오피스텔   

   평수구간  스타일                              세부공사     분야      작업    예산     조회  \
0    35  내추럴                                 0   리모델링     전문가     0  10513   
1    35   모던  주방리모델링, 조명시공, 중문, 가벽&파티션, 슬라이딩도어   리모델링     전문가  4200   5610   
2     5  내추럴                                 0  홈스타일링  셀프•DIY     0   4214   
3    35   모던                  주방리모델링, 조명시공, 중문   리모델링     전문가  5500  37375   
4    55  내추럴                        주방리모델링, 중문   리모델링     전문가  9500   3760   
5    15   모던                                 0  홈스타일링  셀프•DIY     0   8658   
6    35  내추럴                                 0  홈스타일링  셀프•DIY     0   7017   
7    35   모던                              조명시공  홈스타일링  셀프•DIY   500   5620   
8    25   모던                                 0   리모델링     전문가  7000  54268   
9     5  내추럴                                 0  홈스타일링  셀프•DIY   100  72183   

    스크랩  좋아요   댓글                                                url  
0   136   57   15  https://ohou.se/projects/126917/detail?affect_...  
1    75   26   12  https://ohou.se/projects/126600/detail?affect_...  
2   201   60   15  https://ohou.se/projects/123056/detail?affect_...  
3   643  169  122  https://ohou.se/projects/116945/detail?affect_...  
4    73   13    9  https://ohou.se/projects/127436/detail?affect_...  
5   289   79   18  https://ohou.se/projects/125983/detail?affect_...  
6    74   30    6  https://ohou.se/projects/126368/detail?affect_...  
7   100   29    5  https://ohou.se/projects/117185/detail?affect_...  
8  1124  368  116  https://ohou.se/projects/110355/detail?affect_...  
9  3525  725   46  https://ohou.se/projects/100919/detail?affect_...

In [56]:
house_prd = house.merge(prd, how='inner', on ='id')
house_prd.head()

id         작성일                              제목      가족형태   공간  평수구간  스타일  \
0   0  2022-10-07  서재형 거실부터 베란다 아뜰리에까지! 센스만점 오베네홈  아기가 있는 집  아파트    35  내추럴   
1   0  2022-10-07  서재형 거실부터 베란다 아뜰리에까지! 센스만점 오베네홈  아기가 있는 집  아파트    35  내추럴   
2   0  2022-10-07  서재형 거실부터 베란다 아뜰리에까지! 센스만점 오베네홈  아기가 있는 집  아파트    35  내추럴   
3   0  2022-10-07  서재형 거실부터 베란다 아뜰리에까지! 센스만점 오베네홈  아기가 있는 집  아파트    35  내추럴   
4   0  2022-10-07  서재형 거실부터 베란다 아뜰리에까지! 센스만점 오베네홈  아기가 있는 집  아파트    35  내추럴   

  세부공사    분야   작업  ...  댓글                                                url  \
0    0  리모델링  전문가  ...  15  https://ohou.se/projects/126917/detail?affect_...   
1    0  리모델링  전문가  ...  15  https://ohou.se/projects/126917/detail?affect_...   
2    0  리모델링  전문가  ...  15  https://ohou.se/projects/126917/detail?affect_...   
3    0  리모델링  전문가  ...  15  https://ohou.se/projects/126917/detail?affect_...   
4    0  리모델링  전문가  ...  15  https://ohou.se/projects/126917/detail?affect_...   

                                         상품url   브랜드명  \
0   https://ohou.se/productions/538782/selling    에어론   
1   https://ohou.se/productions/803775/selling   베네베네   
2  https://ohou.se/productions/1068049/selling  플렌스테드   
3   https://ohou.se/productions/580038/selling    잠솔솔   
4   https://ohou.se/productions/961600/selling     리스   

                                       상품명                               카테고리  \
0  리버서블 오피스 데스크 책상/테이블 600~2400mm(2colors)  ['가구', '테이블·식탁·책상', '책상', '일반책상']   
1                 벤트라이크 멀티플 접이식 세발자전거+안전가드      ['유아·아동', '완구·교구', '공간·대형완구']   
2    FLENSTED MOBILES 모빌 - 천사 Angel mobile      ['데코·식물', '인테리어소품', '모빌·가랜드']   
3                경추목 메모리폼 숙면 꿀잠 수면 베개_항균탠샐          ['패브릭', '베개·베개커버', '폼베개']   
4                   리스 클래식 법랑 양수냄비 22cm(H)    ['주방용품', '냄비·프라이팬·솥', '냄비·뚝배기']   

       상품가격     리뷰수     스크랩수  스타일링샷수  
0   39800.0  2891.0  57140.0  2113.0  
1   94800.0     1.0     60.0     1.0  
2   37400.0     NaN     81.0     0.0  
3   38900.0    32.0     78.0    13.0  
4  135000.0     1.0     89.0     1.0  

[5 rows x 24 columns]

In [64]:
prd1 = house_prd[:][house_prd['가족형태']=='신혼부부']
prd2 = prd1[:][prd1['스타일']=='내추럴']
prd3 = prd2[:][prd2['스크랩수']>=50]
prd3.head()

id         작성일                              제목  가족형태   공간  평수구간  스타일  \
283   4  2022-10-07  아늑하고 편안한 무드로- 빈티지 한 스푼 더한 내추럴홈  신혼부부  아파트    55  내추럴   
284   4  2022-10-07  아늑하고 편안한 무드로- 빈티지 한 스푼 더한 내추럴홈  신혼부부  아파트    55  내추럴   
285   4  2022-10-07  아늑하고 편안한 무드로- 빈티지 한 스푼 더한 내추럴홈  신혼부부  아파트    55  내추럴   
286   4  2022-10-07  아늑하고 편안한 무드로- 빈티지 한 스푼 더한 내추럴홈  신혼부부  아파트    55  내추럴   
287   4  2022-10-07  아늑하고 편안한 무드로- 빈티지 한 스푼 더한 내추럴홈  신혼부부  아파트    55  내추럴   

           세부공사    분야   작업  ...  댓글  \
283  주방리모델링, 중문  리모델링  전문가  ...   9   
284  주방리모델링, 중문  리모델링  전문가  ...   9   
285  주방리모델링, 중문  리모델링  전문가  ...   9   
286  주방리모델링, 중문  리모델링  전문가  ...   9   
287  주방리모델링, 중문  리모델링  전문가  ...   9   

                                                   url  \
283  https://ohou.se/projects/127436/detail?affect_...   
284  https://ohou.se/projects/127436/detail?affect_...   
285  https://ohou.se/projects/127436/detail?affect_...   
286  https://ohou.se/projects/127436/detail?affect_...   
287  https://ohou.se/projects/127436/detail?affect_...   

                                           상품url      브랜드명  \
283   https://ohou.se/productions/352662/selling      아침향기   
284   https://ohou.se/productions/934004/selling      우정철물   
285   https://ohou.se/productions/311931/selling    우드리퍼블릭   
286  https://ohou.se/productions/1346452/selling   빌레로이앤보흐   
287   https://ohou.se/productions/847875/selling  아메리칸스탠다드   

                                        상품명  \
283            [10%쿠폰]하트아이비_소형기본포트 실내공기정화식물   
284  주물 손잡이 황동 판손잡이 중문 출입문 현관 3colors 3type   
285     트라이탄 양념통 2단 원목 선반세트 중형 (소6P) 2color   
286         [해외] 디자인 나이프 머그 300ml 2P세트 골라담기   
287                             보스톤 원피스 양변기   

                                   카테고리      상품가격   리뷰수   스크랩수 스타일링샷수  
283  ['데코·식물', '플라워·식물', '식물', '모종·씨앗']    3100.0  88.0  166.0   76.0  
284             ['공구·DIY', '손잡이·방문·유리']   35000.0   3.0  169.0    9.0  
285     ['주방용품', '주방수납·정리', '주방선반·정리대']   29800.0  32.0  180.0   26.0  
286          ['주방용품', '컵·잔·텀블러', '머그컵']   54900.0  30.0  240.0   24.0  
287             ['공구·DIY', '수도', '양변기']  775700.0   8.0  343.0   98.0  

[5 rows x 24 columns]

In [65]:
len(prd3['url'].unique())

557

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1554 entries, 5208 to 237877
Data columns (total 24 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   id      1554 non-null   int64  
 1   작성일     1554 non-null   object 
 2   제목      1554 non-null   object 
 3   가족형태    1554 non-null   object 
 4   공간      1554 non-null   object 
 5   평수구간    1554 non-null   int64  
 6   스타일     1554 non-null   object 
 7   세부공사    1554 non-null   object 
 8   분야      1554 non-null   object 
 9   작업      1554 non-null   object 
 10  예산      1554 non-null   int64  
 11  조회      1554 non-null   int64  
 12  스크랩     1554 non-null   int64  
 13  좋아요     1554 non-null   int64  
 14  댓글      1554 non-null   int64  
 15  url     1554 non-null   object 
 16  상품url   1554 non-null   object 
 17  브랜드명    1554 non-null   object 
 18  상품명     1554 non-null   object 
 19  카테고리    1554 non-null   object 
 20  상품가격    1173 non-null   float64
 21  리뷰수     1126 non-null   float64


In [4]:
house_prd = house.merge(prd, how='inner', on='id')
house_prd.head()

id         작성일                              제목      가족형태   공간  평수구간  스타일  \
0   0  2022-10-07  서재형 거실부터 베란다 아뜰리에까지! 센스만점 오베네홈  아기가 있는 집  아파트    35  내추럴   
1   0  2022-10-07  서재형 거실부터 베란다 아뜰리에까지! 센스만점 오베네홈  아기가 있는 집  아파트    35  내추럴   
2   0  2022-10-07  서재형 거실부터 베란다 아뜰리에까지! 센스만점 오베네홈  아기가 있는 집  아파트    35  내추럴   
3   0  2022-10-07  서재형 거실부터 베란다 아뜰리에까지! 센스만점 오베네홈  아기가 있는 집  아파트    35  내추럴   
4   0  2022-10-07  서재형 거실부터 베란다 아뜰리에까지! 센스만점 오베네홈  아기가 있는 집  아파트    35  내추럴   

  세부공사    분야   작업  ...  댓글                                                url  \
0    0  리모델링  전문가  ...  15  https://ohou.se/projects/126917/detail?affect_...   
1    0  리모델링  전문가  ...  15  https://ohou.se/projects/126917/detail?affect_...   
2    0  리모델링  전문가  ...  15  https://ohou.se/projects/126917/detail?affect_...   
3    0  리모델링  전문가  ...  15  https://ohou.se/projects/126917/detail?affect_...   
4    0  리모델링  전문가  ...  15  https://ohou.se/projects/126917/detail?affect_...   

                                         상품url   브랜드명  \
0   https://ohou.se/productions/538782/selling    에어론   
1   https://ohou.se/productions/803775/selling   베네베네   
2  https://ohou.se/productions/1068049/selling  플렌스테드   
3   https://ohou.se/productions/580038/selling    잠솔솔   
4   https://ohou.se/productions/961600/selling     리스   

                                       상품명                               카테고리  \
0  리버서블 오피스 데스크 책상/테이블 600~2400mm(2colors)  ['가구', '테이블·식탁·책상', '책상', '일반책상']   
1                 벤트라이크 멀티플 접이식 세발자전거+안전가드      ['유아·아동', '완구·교구', '공간·대형완구']   
2    FLENSTED MOBILES 모빌 - 천사 Angel mobile      ['데코·식물', '인테리어소품', '모빌·가랜드']   
3                경추목 메모리폼 숙면 꿀잠 수면 베개_항균탠샐          ['패브릭', '베개·베개커버', '폼베개']   
4                   리스 클래식 법랑 양수냄비 22cm(H)    ['주방용품', '냄비·프라이팬·솥', '냄비·뚝배기']   

       상품가격     리뷰수     스크랩수  스타일링샷수  
0   39800.0  2891.0  57140.0  2113.0  
1   94800.0     1.0     60.0     1.0  
2   37400.0     NaN     81.0     0.0  
3   38900.0    32.0     78.0    13.0  
4  135000.0     1.0     89.0     1.0  

[5 rows x 24 columns]

In [8]:
house_prd.columns

Index(['id', '작성일', '제목', '가족형태', '공간', '평수구간', '스타일', '세부공사', '분야', '작업',
       '예산', '조회', '스크랩', '좋아요', '댓글', 'url', '상품url', '브랜드명', '상품명', '카테고리',
       '상품가격', '리뷰수', '스크랩수', '스타일링샷수'],
      dtype='object')

In [9]:
house_prd.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 243019 entries, 0 to 243018
Data columns (total 24 columns):
 #   Column  Non-Null Count   Dtype  
---  ------  --------------   -----  
 0   id      243019 non-null  int64  
 1   작성일     243019 non-null  object 
 2   제목      243019 non-null  object 
 3   가족형태    243019 non-null  object 
 4   공간      243019 non-null  object 
 5   평수구간    243019 non-null  int64  
 6   스타일     243019 non-null  object 
 7   세부공사    243019 non-null  object 
 8   분야      243019 non-null  object 
 9   작업      243019 non-null  object 
 10  예산      243019 non-null  int64  
 11  조회      243019 non-null  int64  
 12  스크랩     243019 non-null  int64  
 13  좋아요     243019 non-null  int64  
 14  댓글      243019 non-null  int64  
 15  url     243019 non-null  object 
 16  상품url   243019 non-null  object 
 17  브랜드명    243019 non-null  object 
 18  상품명     243019 non-null  object 
 19  카테고리    243019 non-null  object 
 20  상품가격    164076 non-null  float64
 21  리뷰수     15

In [ ]:
prd[

#### > 리뷰수, 스크랩수, 스타일링샷수 데이터 타입 변경

house_prd = house_prd.astype({'리뷰수': 'int','스크랩수':'int','스타일링샷수':'int'})

In [5]:
house_prd['id'][house_prd['가족형태']=='신혼부부'].count()

102942

In [50]:
len(house_prd['상품명'].unique())
# 상품 중복 제외 100103개

100103

- 상품 중복 제외 100103개

##### > '신혼부부'로 태깅된 상품 : 102,942개

##### · prd_newlyweds : 가족형태가 신혼부부인 집들이에 태깅된 상품

In [6]:
prd_newlyweds = house_prd[:][house_prd['가족형태']=='신혼부부']
prd_newlyweds.head()

id         작성일                              제목  가족형태   공간  평수구간 스타일  \
100   1  2022-10-07  구축 2Bay 리모델링의 바이블, 모던한 32평 신혼집  신혼부부  아파트    35  모던   
101   1  2022-10-07  구축 2Bay 리모델링의 바이블, 모던한 32평 신혼집  신혼부부  아파트    35  모던   
102   1  2022-10-07  구축 2Bay 리모델링의 바이블, 모던한 32평 신혼집  신혼부부  아파트    35  모던   
103   1  2022-10-07  구축 2Bay 리모델링의 바이블, 모던한 32평 신혼집  신혼부부  아파트    35  모던   
104   1  2022-10-07  구축 2Bay 리모델링의 바이블, 모던한 32평 신혼집  신혼부부  아파트    35  모던   

                                 세부공사    분야   작업  ...  댓글  \
100  주방리모델링, 조명시공, 중문, 가벽&파티션, 슬라이딩도어  리모델링  전문가  ...  12   
101  주방리모델링, 조명시공, 중문, 가벽&파티션, 슬라이딩도어  리모델링  전문가  ...  12   
102  주방리모델링, 조명시공, 중문, 가벽&파티션, 슬라이딩도어  리모델링  전문가  ...  12   
103  주방리모델링, 조명시공, 중문, 가벽&파티션, 슬라이딩도어  리모델링  전문가  ...  12   
104  주방리모델링, 조명시공, 중문, 가벽&파티션, 슬라이딩도어  리모델링  전문가  ...  12   

                                                   url  \
100  https://ohou.se/projects/126600/detail?affect_...   
101  https://ohou.se/projects/126600/detail?affect_...   
102  https://ohou.se/projects/126600/detail?affect_...   
103  https://ohou.se/projects/126600/detail?affect_...   
104  https://ohou.se/projects/126600/detail?affect_...   

                                           상품url   브랜드명  \
100   https://ohou.se/productions/696329/selling   더죤테크   
101  https://ohou.se/productions/1408582/selling  모던하우스   
102  https://ohou.se/productions/1206988/selling   어반테고   
103   https://ohou.se/productions/283304/selling  모던하우스   
104   https://ohou.se/productions/856323/selling  반석스포츠   

                                              상품명  \
100                         제시단101 원홀세면수전 (색상 6종)   
101  ON 높이조절 셀프맞춤 100% 빛차단 반창문 암막커튼 140x180x1P화이트   
102       TET-마시멜로우 MARSH MALLOW 테라조 타일/인테리어 패턴타일   
103               먼지없는 100% 암막커튼 베이지 220x240 2개세트   
104         비스펙 치닝디핑 블랙 턱걸이 철봉 평행봉 풀업운동 헬스기구 운동기구   

                            카테고리      상품가격  리뷰수   스크랩수 스타일링샷수  
100  ['공구·DIY', '수도', '수전·수도용품']  166000.0  1.0  104.0   21.0  
101    ['패브릭', '커튼·부자재', '암막커튼']   23300.0  2.0   95.0    9.0  
102         ['공구·DIY', '타일·파벽돌']   42900.0  NaN  159.0   13.0  
103    ['패브릭', '커튼·부자재', '암막커튼']   89100.0  9.0  183.0   20.0  
104         ['실내운동', '근력', '철봉']   95000.0  8.0  198.0    2.0  

[5 rows x 24 columns]

In [7]:
newlyweds_cg = pd.DataFrame(prd_newlyweds.groupby('카테고리')['상품명'].count())
# newlyweds_cg.sort_values(by='상품명', ascending= False)

# 100번 이상 tagging된 카테고리
newlyweds_cg[:][newlyweds_cg['상품명']>100].sort_values(by='상품명', ascending=False)

상품명
카테고리                                                 
['가구', '의자', '인테리어의자']                           2743
['데코·식물', '플라워·식물', '화병·화분', '화병']               2649
['주방용품', '그릇·홈세트', '접시·플레이트', '일반접시']            1626
['조명', '단스탠드']                                   1610
['가구', '테이블·식탁·책상', '식탁', '식탁·입식테이블']            1519
...                                               ...
['데코·식물', '월데코·장식', '마크라메']                       104
['가구', '행거·옷장', '행거', '고정·설치형행거']                 103
['데코·식물', '데스크·디자인문구', '데스크테리어', '데스크매트·마우스패드']   103
['주방용품', '그릇·홈세트', '접시·플레이트', '찬기']               101
['반려동물', '고양이', '고양이 리빙', '캣타워·캣휠']               101

[235 rows x 1 columns]

## > 카테고리별 상품 평균가격

In [37]:
pd.set_option('display.max_rows',None)

prd_newlyweds_price = pd.DataFrame(prd_newlyweds.groupby('카테고리')['상품가격'].mean())
prd_newlyweds_price.sort_values(by='상품가격',ascending=False)

상품가격
카테고리                                                         
['가구', '침대', '침대+매트리스', '모션베드']                  4.064750e+06
['가전', 'TV', '189cm~(75"~)']                     3.984002e+06
['가전', '생활·건강가전', '안마기·안마의자', '안마기·마사지기']        2.899400e+06
['가전', '냉장고', '양문형냉장고']                          2.834857e+06
['가전', '에어컨', '멀티형']                             2.765907e+06
['인테리어시공', '벽지·장판·마루']                           2.691067e+06
['가전', '세탁기·건조기', '세탁기+건조기']                     2.629319e+06
['가전', '생활·건강가전', '안마기·안마의자', '안마의자']            2.438438e+06
['가구', '침대', '침대+매트리스', '패밀리침대']                 2.430507e+06
['가전', '에어컨', '스탠드형']                            2.418378e+06
['가구', '침대', '침대+매트리스', '일반침대']                  2.332638e+06
['가전', 'TV', '164~188cm(65~74")']                2.276857e+06
['인테리어시공', '주방', '종합시공']                         1.922500e+06
['가전', '컴퓨터·노트북', '노트북', '노트북']                  1.796874e+06
['가전', 'TV', '110~123cm(44~49")']                1.770000e+06
['가전', '컴퓨터·노트북', '컴퓨터', 'PC']                   1.745827e+06
['가구', '침대', '침대+매트리스', '투매트리스']                 1.690330e+06
['가전', '세탁기·건조기', '드럼세탁기']                       1.609753e+06
['가전', '냉장고', '일반냉장고']                           1.603816e+06
['가전', 'TV', '124~163cm(50~64")']                1.585084e+06
['가구', '소파', '일반소파']                             1.570682e+06
['가구', '소파', '리클라이너']                            1.561715e+06
['가전', '에어컨', '냉난방·냉풍기']                         1.536750e+06
['인테리어시공', '창호·폴딩도어']                            1.507500e+06
['가전', '냉장고', '김치냉장고']                           1.489916e+06
['가전', '세탁기·건조기', '의류관리기']                       1.460163e+06
['가전', '디지털가전', '휴대폰']                           1.363200e+06
['가전', '세탁기·건조기', '건조기·탈수기']                     1.325989e+06
['가전', '주방가전', '식기세척기·건조기', '식기세척기']             1.304286e+06
['가구', '의자', '안락의자']                             1.244337e+06
['가구', '매트리스·토퍼', '매트리스', '스프링매트리스']             1.177724e+06
['가전', '음향가전', '스피커', '일반스피커']                   1.151343e+06
['가전', 'TV', '~109cm(~43")']                     1.147203e+06
['가전', '주방가전', '가스레인지·인덕션', '인덕션']               1.114037e+06
['가전', '냉장고', '냉동고']                             1.100331e+06
['가전', '영상가전', '빔프로젝터']                          1.077431e+06
['가전', '디지털가전', '태블릿PC']                         1.014452e+06
['가구', '침대', '침대프레임', '모션베드']                    1.010380e+06
['가전', '주방가전', '정수기']                            9.792354e+05
['인테리어시공', '중문·도어']                              9.727938e+05
['가전', '에어컨', '창문·이동형']                          8.663333e+05
['가구', '침대', '침대프레임', '패밀리침대']                   8.419967e+05
['가구', '침대', '침대프레임', '일반침대']                    8.323781e+05
['가전', '에어컨', '천장형']                             7.900000e+05
['가구', '매트리스·토퍼', '매트리스', '폼매트리스']               7.855357e+05
['캠핑용품', '텐트·타프', '텐트']                          7.761948e+05
['가전', '음향가전', '스피커', '사운드바']                    7.499828e+05
['가구', '침대', '침대프레임', '수납침대']                    7.492503e+05
['가구', '테이블·식탁·책상', '식탁', '식탁+의자']               7.395139e+05
['가전', '주방가전', '가스레인지·인덕션', '하이브리드']             7.158462e+05
['가구', '행거·옷장', '붙박이장']                          7.157876e+05
['가구', '테이블·식탁·책상', '책상', '학원·사무용책상']            6.971000e+05
['가전', '이미용가전', '세팅롤·브러쉬']                       6.920414e+05
['공구·DIY', '수도', '양변기']                          6.833032e+05
['가구', '거실장·TV장', '높은거실장·사이드보드']                 6.731879e+05
['가전', '에어컨', '벽걸이형']                            6.689176e+05
['가구', '침대', '침대+매트리스', '수납침대']                  6.683700e+05
['가전', '주방가전', '오븐·전자레인지', '오븐']                 6.330758e+05
['가전', '주방가전', '오븐·전자레인지', '오븐레인지']              6.293333e+05
['가전', '주방가전', '음식물처리기']                         6.285119e+05
['가전', '냉장고', '미니냉장고']                           6.186875e+05
['가전', '청소기', '로봇청소기']                           6.145441e+05
['가전', '청소기', '일반청소기']                           6.133107e+05
['가전', '주방가전', '커피메이커·머신', 

### > 문제점

- 신혼부부 페이지 태깅 상품들의 카테고리 그룹해서 평균가격 구하는 것에 대함.
- 평균 가격대는 상품 특성 대비 어느정도 맞는 것 같음
- 추천 시스템에서 '선물하기'로 제한한다면 가격 하,상한선이 필요할 것 같음.

Index(['id', '작성일', '제목', '가족형태', '공간', '평수구간', '스타일', '세부공사', '분야', '작업',\
　　　'예산', '조회', '스크랩', '좋아요', '댓글', 'url', '상품url', '브랜드명', '상품명', '카테고리',\
　　　'상품가격', '리뷰수', '스크랩수', '스타일링샷수'],\
      dtype='object')

## > 스타일별 상품 개수

In [17]:
# 모던
prd_modern = prd_newlyweds[:][prd_newlyweds['스타일']=='모던']
prd_modern.head()  # 19463개

id         작성일                              제목  가족형태   공간  평수구간 스타일  \
100   1  2022-10-07  구축 2Bay 리모델링의 바이블, 모던한 32평 신혼집  신혼부부  아파트    35  모던   
101   1  2022-10-07  구축 2Bay 리모델링의 바이블, 모던한 32평 신혼집  신혼부부  아파트    35  모던   
102   1  2022-10-07  구축 2Bay 리모델링의 바이블, 모던한 32평 신혼집  신혼부부  아파트    35  모던   
103   1  2022-10-07  구축 2Bay 리모델링의 바이블, 모던한 32평 신혼집  신혼부부  아파트    35  모던   
104   1  2022-10-07  구축 2Bay 리모델링의 바이블, 모던한 32평 신혼집  신혼부부  아파트    35  모던   

                                 세부공사    분야   작업  ...  댓글  \
100  주방리모델링, 조명시공, 중문, 가벽&파티션, 슬라이딩도어  리모델링  전문가  ...  12   
101  주방리모델링, 조명시공, 중문, 가벽&파티션, 슬라이딩도어  리모델링  전문가  ...  12   
102  주방리모델링, 조명시공, 중문, 가벽&파티션, 슬라이딩도어  리모델링  전문가  ...  12   
103  주방리모델링, 조명시공, 중문, 가벽&파티션, 슬라이딩도어  리모델링  전문가  ...  12   
104  주방리모델링, 조명시공, 중문, 가벽&파티션, 슬라이딩도어  리모델링  전문가  ...  12   

                                                   url  \
100  https://ohou.se/projects/126600/detail?affect_...   
101  https://ohou.se/projects/126600/detail?affect_...   
102  https://ohou.se/projects/126600/detail?affect_...   
103  https://ohou.se/projects/126600/detail?affect_...   
104  https://ohou.se/projects/126600/detail?affect_...   

                                           상품url   브랜드명  \
100   https://ohou.se/productions/696329/selling   더죤테크   
101  https://ohou.se/productions/1408582/selling  모던하우스   
102  https://ohou.se/productions/1206988/selling   어반테고   
103   https://ohou.se/productions/283304/selling  모던하우스   
104   https://ohou.se/productions/856323/selling  반석스포츠   

                                              상품명  \
100                         제시단101 원홀세면수전 (색상 6종)   
101  ON 높이조절 셀프맞춤 100% 빛차단 반창문 암막커튼 140x180x1P화이트   
102       TET-마시멜로우 MARSH MALLOW 테라조 타일/인테리어 패턴타일   
103               먼지없는 100% 암막커튼 베이지 220x240 2개세트   
104         비스펙 치닝디핑 블랙 턱걸이 철봉 평행봉 풀업운동 헬스기구 운동기구   

                            카테고리      상품가격  리뷰수   스크랩수 스타일링샷수  
100  ['공구·DIY', '수도', '수전·수도용품']  166000.0  1.0  104.0   21.0  
101    ['패브릭', '커튼·부자재', '암막커튼']   23300.0  2.0   95.0    9.0  
102         ['공구·DIY', '타일·파벽돌']   42900.0  NaN  159.0   13.0  
103    ['패브릭', '커튼·부자재', '암막커튼']   89100.0  9.0  183.0   20.0  
104         ['실내운동', '근력', '철봉']   95000.0  8.0  198.0    2.0  

[5 rows x 24 columns]

In [21]:
prd_modern['카테고리'].value_counts().head(50)

['가구', '의자', '인테리어의자']                    459
['데코·식물', '플라워·식물', '화병·화분', '화병']        418
['공구·DIY', '전기', '스위치·콘센트']               378
['주방용품', '그릇·홈세트', '접시·플레이트', '일반접시']     341
['조명', '단스탠드']                            281
['가구', '테이블·식탁·책상', '식탁', '식탁·입식테이블']     265
['조명', '천장등', '펜던트조명']                    259
['가구', '소파', '일반소파']                      244
['공구·DIY', '수도', '수전·수도용품']               241
['데코·식물', '플라워·식물', '화병·화분', '화분']        229
['공구·DIY', '손잡이·방문·유리']                   206
['가구', '의자', '스툴·벤치']                     186
['데코·식물', '홈갤러리', '액자']                   180
['가전', '음향가전', '스피커', '블루투스·인공지능스피커']     179
['수납·정리', '화장대·테이블정리', '티슈커버·케이스']        174
['생활용품', '욕실용품', '욕실선반·욕실정리']             173
['주방용품', '컵·잔·텀블러', '맥주잔']                165
['주방용품', '컵·잔·텀블러', '와인·양주·칵테일잔']         163
['주방용품', '컵·잔·텀블러', '머그컵']                163
['가구', '테이블·식탁·책상', '사이드테이블']             163
['조명', '장스탠드']                            160
['가전', '주방가전', '커피메이커·머신', '캡슐커피머신

In [22]:
# 내추럴
prd_natural = prd_newlyweds[:][prd_newlyweds['스타일']=='내추럴']
prd_natural.head()  # 19463개

id         작성일                              제목  가족형태   공간  평수구간  스타일  \
283   4  2022-10-07  아늑하고 편안한 무드로- 빈티지 한 스푼 더한 내추럴홈  신혼부부  아파트    55  내추럴   
284   4  2022-10-07  아늑하고 편안한 무드로- 빈티지 한 스푼 더한 내추럴홈  신혼부부  아파트    55  내추럴   
285   4  2022-10-07  아늑하고 편안한 무드로- 빈티지 한 스푼 더한 내추럴홈  신혼부부  아파트    55  내추럴   
286   4  2022-10-07  아늑하고 편안한 무드로- 빈티지 한 스푼 더한 내추럴홈  신혼부부  아파트    55  내추럴   
287   4  2022-10-07  아늑하고 편안한 무드로- 빈티지 한 스푼 더한 내추럴홈  신혼부부  아파트    55  내추럴   

           세부공사    분야   작업  ...  댓글  \
283  주방리모델링, 중문  리모델링  전문가  ...   9   
284  주방리모델링, 중문  리모델링  전문가  ...   9   
285  주방리모델링, 중문  리모델링  전문가  ...   9   
286  주방리모델링, 중문  리모델링  전문가  ...   9   
287  주방리모델링, 중문  리모델링  전문가  ...   9   

                                                   url  \
283  https://ohou.se/projects/127436/detail?affect_...   
284  https://ohou.se/projects/127436/detail?affect_...   
285  https://ohou.se/projects/127436/detail?affect_...   
286  https://ohou.se/projects/127436/detail?affect_...   
287  https://ohou.se/projects/127436/detail?affect_...   

                                           상품url      브랜드명  \
283   https://ohou.se/productions/352662/selling      아침향기   
284   https://ohou.se/productions/934004/selling      우정철물   
285   https://ohou.se/productions/311931/selling    우드리퍼블릭   
286  https://ohou.se/productions/1346452/selling   빌레로이앤보흐   
287   https://ohou.se/productions/847875/selling  아메리칸스탠다드   

                                        상품명  \
283            [10%쿠폰]하트아이비_소형기본포트 실내공기정화식물   
284  주물 손잡이 황동 판손잡이 중문 출입문 현관 3colors 3type   
285     트라이탄 양념통 2단 원목 선반세트 중형 (소6P) 2color   
286         [해외] 디자인 나이프 머그 300ml 2P세트 골라담기   
287                             보스톤 원피스 양변기   

                                   카테고리      상품가격   리뷰수   스크랩수 스타일링샷수  
283  ['데코·식물', '플라워·식물', '식물', '모종·씨앗']    3100.0  88.0  166.0   76.0  
284             ['공구·DIY', '손잡이·방문·유리']   35000.0   3.0  169.0    9.0  
285     ['주방용품', '주방수납·정리', '주방선반·정리대']   29800.0  32.0  180.0   26.0  
286          ['주방용품', '컵·잔·텀블러', '머그컵']   54900.0  30.0  240.0   24.0  
287             ['공구·DIY', '수도', '양변기']  775700.0   8.0  343.0   98.0  

[5 rows x 24 columns]

In [23]:
prd_natural['카테고리'].value_counts().head(50)

['데코·식물', '플라워·식물', '화병·화분', '화병']        923
['가구', '의자', '인테리어의자']                    895
['주방용품', '그릇·홈세트', '접시·플레이트', '일반접시']     593
['데코·식물', '플라워·식물', '화병·화분', '화분']        544
['조명', '단스탠드']                            542
['가구', '테이블·식탁·책상', '식탁', '식탁·입식테이블']     497
['조명', '천장등', '펜던트조명']                    464
['가구', '소파', '일반소파']                      463
['공구·DIY', '전기', '스위치·콘센트']               461
['가구', '의자', '스툴·벤치']                     414
['가구', '서랍·수납장', '수납장']                   373
['데코·식물', '인테리어소품', '장식소품']               357
['데코·식물', '홈갤러리', '액자']                   352
['가구', '테이블·식탁·책상', '사이드테이블']             339
['가전', '음향가전', '스피커', '블루투스·인공지능스피커']     317
['공구·DIY', '수도', '수전·수도용품']               316
['패브릭', '쿠션·방석', '쿠션']                    303
['주방용품', '컵·잔·텀블러', '머그컵']                284
['수납·정리', '화장대·테이블정리', '티슈커버·케이스']        275
['가전', '주방가전', '커피메이커·머신', '캡슐커피머신']      268
['가구', '서랍·수납장', '서랍장']                   263
['데코·식물', '시계', '벽시계']            

In [40]:
pd.set_option('display.max_rows',None)

# prd_cd : 스타일별 상품 카테고리 분류 그룹
prd_cd = pd.DataFrame(prd_newlyweds['스타일'].value_counts())
prd_cd = prd_cd.reset_index().sort_values(by='index').rename(columns={'index':'스타일','스타일':'상품갯수'})

In [42]:
prd_cd

스타일   상품갯수
2                                               0  12811
0                                             내추럴  35801
24                                   내추럴, 러블리&로맨틱    292
68                         내추럴, 러블리&로맨틱, 유니크&믹스매치     56
11                                       내추럴, 북유럽    902
40                              내추럴, 북유럽, 러블리&로맨틱    118
51                              내추럴, 북유럽, 빈티지&레트로     96
41                     내추럴, 북유럽, 빈티지&레트로, 러블리&로맨틱    115
89                    내추럴, 북유럽, 빈티지&레트로, 유니크&믹스매치     36
70                      내추럴, 북유럽, 빈티지&레트로, 클래식&앤틱     55
73                      내추럴, 북유럽, 빈티지&레트로, 한국&아시아     51
25                             내추럴, 북유럽, 유니크&믹스매치    249
92                               내추럴, 북유럽, 클래식&앤틱     34
9                                    내추럴, 빈티지&레트로   1764
47                          내추럴, 빈티지&레트로, 러블리&로맨틱    101
69                         내추럴, 빈티지&레트로, 유니크&믹스매치     55
42                           내추럴, 빈티지&레트로, 클래식&앤틱    114
95                  내추럴, 빈티지&레트로, 클래식&앤틱, 러블리&로맨틱     33
55         내추럴, 빈티지&레트로, 클래식&앤틱, 프렌치&프로방스, 한국&아시아     89
108                  내추럴, 빈티지&레트로, 클래식&앤틱, 한국&아시아     13
82                 내추럴, 빈티지&레트로, 한국&아시아, 유니크&믹스매치     41
32                                  내추럴, 유니크&믹스매치    192
98                                    내추럴, 클래식&앤틱     31
107                          내추럴, 클래식&앤틱, 러블리&로맨틱     17
45                                  내추럴, 프렌치&프로방스    111
50                                    내추럴, 한국&아시아     97
15                                        러블리&로맨틱    513
78                              러블리&로맨틱, 유니크&믹스매치     48
1                                              모던  19463
8                                         모던, 내추럴   1783
34                                   모던, 내추럴, 북유럽    189
80                          모던, 내추럴, 북유럽, 러블리&로맨틱     44
63                모던, 내추럴, 북유럽, 빈티지&레트로, 유니크&믹스매치     62
86                         모던, 내추럴, 북유럽, 프렌치&프로방스     37
36                               모던, 내추럴, 빈티지&레트로    143
53                     모던, 내추럴, 빈티지&레트로, 유니크&믹스매치     90
71                       모던, 내추럴, 빈티지&레트로, 클래식&앤틱     52
54                              모던, 내추럴, 유니크&믹스매치     89
67                                모던, 내추럴, 클래식&앤틱     56
75   모던, 내추럴, 클래식&앤틱, 프렌치&프로방스, 러블리&로맨틱, 유니크&믹스매치     48
52                                    모던, 러블리&로맨틱     94
43                          모던, 러블리&로맨틱, 유니크&믹스매치    113
5                                      모던, 미니멀&심플   2897
6                                 모던, 미니멀&심플, 내추럴   2538
60                       모던, 미니멀&심플, 내추럴, 러블리&로맨틱     67
26                           모던, 미니멀&심플, 내추럴, 북유럽    236
74                       모던, 미니멀&심플, 내추럴, 빈티지&레트로     50
100            모던, 미니멀&심플, 내추럴, 빈티지&레트로, 유니크&믹스매치     28
39               모던, 미니멀&심플, 내추럴, 빈티지&레트로, 클래식&앤틱    119
62               모던, 미니멀&심플, 내추럴, 빈티지&레트로, 한국&아시아     63
27                        모던, 미니멀&심플, 내추럴, 클래식&앤틱    235
59                            모던, 미니멀&심플, 러블리&로맨틱     69
29                                모던, 미니멀&심플, 북유럽    210
20                            모던, 미니멀&심플, 빈티지&레트로    341
91                           모던, 미니멀&심플, 유니크&믹스매치     36
85                             모던, 미니멀&심플, 클래식&앤틱     39
106                          모던, 미니멀&심플, 프렌치&프로방스     19
19                                        모던, 북유럽    372
101                    모던, 북유럽, 빈티지&레트로, 유니크&믹스매치     28
57                              모던, 북유럽, 유니크&믹스매치     74
87                                모던, 북유럽, 인더스트리얼     37
18                                    모던, 빈티지&레트로    375
81                    모던, 빈티지&레트로, 클래식&앤틱, 인더스트리얼     43
64                  모던, 빈티지&레트로, 클래식&앤틱, 프렌치&프로방스     59
30                                   모던, 유니크&믹스매치    199
65                            모던, 클래식&앤틱, 러블리&로맨틱     58
4                                          미니멀&심플   3471
3                                     미니멀&심플, 내추럴   5152
37                           미니멀&심플, 내추럴, 러블리&로맨틱    141
14                               미니멀&심플, 내추럴, 북유럽    538

In [92]:
prd_cd['상품갯수'][prd_cd['스타일'].apply(lambda x : '내추럴' in x)].sum()

52864

In [93]:
prd_cd['상품갯수'][prd_cd['스타일'].apply(lambda x : '모던' in x)].sum()

30455

In [96]:
prd_cd['상품갯수'][prd_cd['스타일'].apply(lambda x : '인더스트리얼' in x)].sum()

263

- 스타일 null값 = 12811개
#### 조인하고 추천에서는 제거 필요할 것 같음.

##  > 각 카테고리를 포함하는 상품갯수의 비율 ?

In [86]:
style = ['내추럴', '러블리&로맨틱', '북유럽', '빈티지&레트로', '유니크&믹스매치', '클래식&앤틱', '한국&아시아', '프렌치&프로방스','모던', '미니멀&심플', '인더스트리얼']

cnt = 0
cmt = 0
for i in style:
    for j in range(0,109):
        if i in prd_cd.iloc[j,0]:
            # prd_cd[:][prd_cd['스타일'].apply(lambda x : i in x)]:
            cnt = cnt + prd_cd.iloc[j,1]
        
        else:
            pass
        
    print(i, cnt)
    cmt += cmt + cnt
print(cmt)

내추럴 52864
러블리&로맨틱 55239
북유럽 60607
빈티지&레트로 69161
유니크&믹스매치 73016
클래식&앤틱 75013
한국&아시아 76112
프렌치&프로방스 77212
모던 107667
미니멀&심플 125491
인더스트리얼 125754
116499436


### > 각 단일 스타일별 포함하는 상품갯수 비율

In [95]:
style = ['내추럴', '러블리&로맨틱', '북유럽', '빈티지&레트로', '유니크&믹스매치', '클래식&앤틱', '한국&아시아', '프렌치&프로방스','모던', '미니멀&심플', '인더스트리얼']

cnt = 0
cmt = 0
for i in style:
    for j in range(0,109):
        if i in prd_cd.iloc[j,0]:
            # prd_cd[:][prd_cd['스타일'].apply(lambda x : i in x)]:
            cnt = cnt + prd_cd.iloc[j,1]
        
        else:
            pass
        
    print(i, cnt, round(cnt/116499436 * 100, 4))
    cmt += cmt + cnt
print(cmt)

내추럴 52864 0.0454
러블리&로맨틱 55239 0.0474
북유럽 60607 0.052
빈티지&레트로 69161 0.0594
유니크&믹스매치 73016 0.0627
클래식&앤틱 75013 0.0644
한국&아시아 76112 0.0653
프렌치&프로방스 77212 0.0663
모던 107667 0.0924
미니멀&심플 125491 0.1077
인더스트리얼 125754 0.1079
116499436


#### → 상품이 가장 많이 태깅된 스타일 = '인더스트리얼'
#### → 상품이 가장 적게 태깅된 스타일 = '내추럴'

In [51]:
prd_natural['카테고리'].value_counts()

['데코·식물', '플라워·식물', '화병·화분', '화병']                 923
['가구', '의자', '인테리어의자']                             895
['주방용품', '그릇·홈세트', '접시·플레이트', '일반접시']              593
['데코·식물', '플라워·식물', '화병·화분', '화분']                 544
['조명', '단스탠드']                                     542
['가구', '테이블·식탁·책상', '식탁', '식탁·입식테이블']              497
['조명', '천장등', '펜던트조명']                             464
['가구', '소파', '일반소파']                               463
['공구·DIY', '전기', '스위치·콘센트']                        461
['가구', '의자', '스툴·벤치']                              414
['가구', '서랍·수납장', '수납장']                            373
['데코·식물', '인테리어소품', '장식소품']                        357
['데코·식물', '홈갤러리', '액자']                            352
['가구', '테이블·식탁·책상', '사이드테이블']                      339
['가전', '음향가전', '스피커', '블루투스·인공지능스피커']              317
['공구·DIY', '수도', '수전·수도용품']                        316
['패브릭', '쿠션·방석', '쿠션']                             303
['주방용품', '컵·잔·텀블러', '머그컵']                         284
['수납·정리', 

In [54]:
# 빈티지
prd_vintage = prd_newlyweds[:][prd_newlyweds['스타일']=='빈티지&레트로']
prd_vintage['카테고리'].value_counts()

['데코·식물', '플라워·식물', '화병·화분', '화분']                68
['가구', '의자', '인테리어의자']                            60
['데코·식물', '플라워·식물', '화병·화분', '화병']                45
['조명', '단스탠드']                                    43
['가구', '서랍·수납장', '수납장']                           38
['조명', '천장등', '펜던트조명']                            37
['주방용품', '그릇·홈세트', '접시·플레이트', '일반접시']             31
['가구', '소파', '일반소파']                              28
['가구', '진열장·책장', '진열장·장식장']                       27
['가구', '거울', '벽거울']                               27
['가구', '테이블·식탁·책상', '식탁', '식탁·입식테이블']             27
['주방용품', '컵·잔·텀블러', '맥주잔']                        27
['데코·식물', '홈갤러리', '액자']                           25
['데코·식물', '홈갤러리', '그림·사진', '일러스트·드로잉']            24
['가구', '의자', '스툴·벤치']                             23
['패브릭', '커튼·부자재', '가리개커튼']                        23
['패브릭', '쿠션·방석', '쿠션']                            23
['가전', '음향가전', '스피커', '블루투스·인공지능스피커']             22
['주방용품', '컵·잔·텀블러', '머그컵']                    

In [55]:
# 믹스매치
prd_unique = prd_newlyweds[:][prd_newlyweds['스타일']=='유니크&믹스매치']
prd_unique['카테고리'].value_counts()

['가구', '의자', '인테리어의자']                             40
['데코·식물', '플라워·식물', '화병·화분', '화병']                 37
['패브릭', '베개·베개커버', '베개커버']                         35
['조명', '단스탠드']                                     33
['주방용품', '그릇·홈세트', '접시·플레이트', '일반접시']              27
['가전', '음향가전', '스피커', '블루투스·인공지능스피커']              24
['가구', '테이블·식탁·책상', '식탁', '식탁·입식테이블']              24
['가구', '의자', '스툴·벤치']                              22
['조명', '장스탠드']                                     20
['가구', '소파', '일반소파']                               20
['패브릭', '이불·이불솜', '이불', '이불커버']                    19
['조명', '천장등', '펜던트조명']                             18
['가구', '서랍·수납장', '수납장']                            18
['가구', '테이블·식탁·책상', '사이드테이블']                      18
['데코·식물', '플라워·식물', '화병·화분', '화분']                 17
['수납·정리', '화장대·테이블정리', '티슈커버·케이스']                 17
['가구', '테이블·식탁·책상', '책상', '일반책상']                  16
['패브릭', '커튼·부자재', '가리개커튼']                         15
['패브릭', '매트리스커버']           

**[니즈]** 집들이 선물하려고 오늘의 집에 들어왔는데 구매할 상품을 결정하는 데 어려움이 있다.

**[가설]**

- ****집들이 데이터를 기반으로 **‘집들이’라는 목적**에 맞게 카테고리를 세분화할 수 있을 것이다.

        (가구형태, 분위기, 공간, 취향, 예산에 따라)

- 추천 탭을 재구성한다면 추천 시스템에 대한 소비자의 만족도가 올라갈 것이다.

**[문제점 및 해결방안]**

- 상품을 정하지 않고 물건을 고를 때 (ex 원하는 분위기의 상품을 찾을 때, 신혼부부에게 좋은 선물을 찾을 때)는 기존 필터가 도움이 안된다.

- **새로운 필터 생성**
    - 예시) 우리가 새로 만든 카테고리(가구형태, 분위기, 공간, 취향, 예산)로 필터링된 상품 리스트
    - 
- 현재 추천 탭 분류 기준이 모호하고 추천 상품들이 너무 많다.
    - **추천 탭 재구성**
        - 예시) 함께 꾸미기 좋은 상품 (스타일 기반)
                